code/ui/Nameplate.cs

```diff
@@ -9,15 +9,68 @@ namespace TTTReborn.UI
 {
     public class Nameplate : Panel
     {
+        public static Nameplate Instance;
+
+        public bool IsShowing
+        {
+            get => _isShowing;
+            set
+            {
+                _isShowing = value;
+
+                SetClass("hide", !_isShowing);
+            }
+        }
+
+        private bool _isShowing = false;
+
         private const float MAX_DRAW_DISTANCE = 500;
 
+        private readonly Panel _labelHolder;
+        private readonly Panel _nameHolder;
+
         private readonly Label _nameLabel;
+        private readonly Label _damageIndicatorLabel;
+
+        private float _playerHp;
 
         public Nameplate()
         {
+            Instance = this;
+            IsShowing = false;
+
             StyleSheet.Load("/ui/Nameplate.scss");
 
-            _nameLabel = Add.Label("", "name");
+            _labelHolder = Add.Panel("labelHolder");
+
+            _nameHolder = _labelHolder.Add.Panel("nameHolder");
+            _nameLabel = _nameHolder.Add.Label("", "name");
+
+            _damageIndicatorLabel = _labelHolder.Add.Label("", "damageIndicator");
+
+        }
+
+        private string GetHealthGroup(float health)
+        {
+            return health > 70 ? "Healthy"
+                : health > 20 ? "Injured"
+                : "Near death";
```

Comment:
Same as other comment, we should const these magic numbers. And even potentially the strings themself.